partial-flashing.js

```diff
@@ -698,14 +700,19 @@ let PartialFlashing = {
                 });
 
             let timeout = new Promise((resolve, reject) => {
-                setTimeout(() => {
+                resetTimeout = setTimeout(() => {
+                    PartialFlashingUtils.log("Resetting micro:bit timed out");
                     reject('Timeout')
                 }, 1000)
             })
 
             // Use race to timeout the reset.
             let ret = await Promise.race([p, timeout])
```

Comment:
Based on this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race

It sounds like if the timeout triggers a reject, and there is not a `catch` in this promise chain, then the operation would just disconnect, as part of the `finally()`, but nothing else would happen. 
The "full flashing" back up wouldn't be attempted in this case, as it is executed as part of a try/catch, and this catch will only be triggered if an exception is thrown, and this is only triggering a `reject()`, no?

Reply:
I think it is attempted, as it's all contained within the `try { .. } catch { .. }`

It's probably best to rewrite the race so that it checks which promise has completed first and either continue parital flash or start full flash, rather than rely on the reject being caught

Reply:
> I think it is attempted, as it's all contained within the try { .. } catch { .. }

But a reject is not "caught" in a `try { .. } catch { .. }`, right? It's just exceptions that are caught?
If you think there is a best way to rewrite this then we should probably do that.

Reply:
I've modified it so both promises resolve and both paths are implemented in the `Promise.race`